lib/MemoryHandler.js

```diff
@@ -80,9 +92,14 @@ MemoryStore.prototype.delete = function(request, callback) {
   this.find(request, function(err, theResource) {
     if (err) return callback(err);
 
-    // Remove the resource from the in-meory store.
-    var resourceIndex = resources[request.params.type].indexOf(theResource);
-    resources[request.params.type].splice(resourceIndex, 1);
+    // Remove the resource from the in-memory store.
+    resources[request.params.type].some(function(resource, index) {
+      var match = resource.id === theResource.id;
+      if (match) {
+        resources[request.params.type].splice(index, 1);
+      }
+      return match;
+    });
```

Comment:
I shall note the clever use of `Array.some` while at the same time lamenting the regrettable side effect in its callback. Since [`Array.prototype.findIndex`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) is ES6 only I guess this is as good as the alternative approaches.